types/react-router/index.d.ts

```diff
@@ -148,3 +148,13 @@ export function withRouter<P extends RouteComponentProps<any>, C extends React.C
 ): React.ComponentClass<Omit<P, keyof RouteComponentProps<any>> & WithRouterProps<C>> & WithRouterStatics<C>;
 
 export const __RouterContext: React.Context<RouteComponentProps>;
+
+export function useHistory(): H.History;
```

Comment:
Can we include a type param for the history state `S` here?

Reply:
I went ahead and named the type param `HistoryLocationState` to match the definition for `H.History`

Reply:
Nice